data/data_manager.py

```diff
@@ -67,29 +67,33 @@ def _append_csv_data(self):
         '''
         
         # Find all data keys
-        all_keys = []
+        all_keys = [] # All data keys
+        zon_keys = [] # Subset of data keys that allow a zone specifier 
+        
         for category in self.categories:
             all_keys.extend(self.categories[category])
+            if category in ['occupancy','internalGains','setpoints']:
+                zon_keys.extend(self.categories[category])
         
         # Keep track of the data already appended to avoid duplication
-        appended = {key: None for key in all_keys}
+        appended = {}
         
         # Search for .csv files in the resources folder
         for f in self.files:
             if f.endswith('.csv'):
                 df = pd.read_csv(f, comment='#')
-                keys = df.keys()
-                if 'time' in keys:
-                    for key in keys.drop('time'):
-                        # Raise error if key already appended
-                        if appended[key] is not None:
-                            raise ReferenceError('{0} in multiple files within the Resources folder. These are: {1}, and {2}'.format(key, appended[key], f))
-                        # Trim df from keys that are not in categories
-                        elif key not in all_keys:
-                            df.drop(key, inplace=True)
+                cols = df.keys()
+                if 'time' in cols:
+                    for col in cols.drop('time'):
+                        # Raise error if col already appended
+                        if col in appended.keys():
+                            raise ReferenceError('{0} in multiple files within the Resources folder. These are: {1}, and {2}'.format(col, appended[col], f))
+                        # Trim df from cols that are not in categories
+                        elif not( (col in all_keys) or any(col.startswith(key) for key in zon_keys) ):
```

Comment:
@JavierArroyoBastida Should there be a stronger check of the format ``'<category>[<zone>]'``?  In https://github.com/ibpsa/project1-boptest/pull/169/commits/26d89cf9d0d97985eaf034d9b7849e198e673d50 line 111-112 there is a check, but would be worth having when the test case is being compiled, instead of waiting until trying to calculate kpis.

Reply:
@dhblum I agree it's a good idea to have the check when the test is being compiled. Shall I then remove the check in the KPI calculator? I cannot think of a case where the check in the KPI calculator could be useful if the check in the data manager has already filtered out the wrong keys.

Reply:
I would keep the check in the KPI calculator too, in the case the test case fmu is not compiled using the data manager.

Reply:
What if we include the stronger check also when loading the test case with the data? so include it in:
https://github.com/ibpsa/project1-boptest/blob/f985196de852d8b833e2a56d12a8a11b5eae0650/data/data_manager.py#L247
This way the test case data will always undergo the test independently of how it was compiled.  In that case, wouldn't it be the checks at KPI Calculator redundant? We can always keep them but I think it's better if we can avoid them as KPI Calculator could be called iteratively if the user wants to use it on the fly.

Reply:
Yes, you're right, checking when loading the data would be best.  It should not be needed in the KPI Calc then.